foreshadow/intents/general.py

```diff
@@ -66,6 +92,55 @@ def is_intent(cls, df):
             .all()
         )
 
+    @classmethod
+    def column_summary(cls, df):
+        """Returns computed statistics for a NumericIntent column
+
+            The following are computed:
+                nan: count of nans pass into dataset
+                invalid: number of invalid values after converting to numeric
+                mean: -
+                std: -
+                min: -
+                25th: 25th percentile
+                median: -
+                75th: 75th percentile
+                max: -
+                mode: mode or np.nan if data is mostly unique
+                top10: top 10 most frequent values or empty array if mostly unique
+                    [(value, count),...,]
+                10outliers: largest 10 outliers
+
+        """
+        data = df.ix[:, 0]
+        nan_num = int(data.isnull().sum())
+        invalid_num = int(
+            pd.to_numeric(df.ix[:, 0], errors="coerce").isnull().sum() - nan_num
+        )
+        outliers = (
+            data[np.abs(data - data.mean()) > (3 * data.std())]
+            .nlargest(10)
```

Comment:
this will only take the positive side, right?

Reply:
that's a great catch